src/arch/timerha.cpp

```diff
@@ -82,13 +83,11 @@ void CTimerHandler::nextTick() {
 }
 
 void CTimerHandler::processTimedFBList() {
-  for (auto it = mTimedFBList.begin(); it != mTimedFBList.end();) {
-    if (it->mTimeOut > mForteTime) {
-      break;  
-    }
-    STimedFBListEntry entry = *it; // buffer entry
-    it = mTimedFBList.erase(it);
-    triggerTimedFB(entry);
+  auto last = std::lower_bound(mTimedFBList.begin(), mTimedFBList.end(), mForteTime);
```

Comment:
This is not equivalent. It leaves values that are equal to the target value in the list, whereas before equal values would be removed. The solution is to use `std::upper_bound`.

Reply:
Thx for pointing this out. Changed that.